cumulus_library/studies/core/builder_documentreference.py

```diff
@@ -35,7 +36,31 @@ def prepare_queries(
                     column_name="type",
                     is_array=False,
                     target_table="core__documentreference_dn_type",
-                )
+                ),
+                # TODO: The spec allows any value for category, but it's unclear what
+                # the value for the systems would be. For the time being, we select the
+                # spec valid code (which is a set of one value).
+                # It may be worth confirming the values in this field with either the
+                # quality or discovery studies on an ongoing basis to find other valid
+                # values
+                sql_utils.CodeableConceptConfig(
+                    source_table="documentreference",
+                    source_id="id",
+                    column_name="category",
+                    is_array=True,
+                    filter_priority=True,
+                    target_table="core__documentreference_dn_category",
+                    code_systems=[
+                        "http://hl7.org/fhir/us/core/ValueSet/us-core-documentreference-category"
+                    ],
+                ),
+                sql_utils.CodeableConceptConfig(
+                    source_table="documentreference",
+                    source_id="id",
+                    column_name="content.format",
+                    is_array=False,
```

Comment:
`content` is an array though - does the generator automatically assume parents are arrays?

Reply:
good catch - will update.

Reply:
for posterity - I think this actually requires a larger chunk of effort, since this is a coding rather than a codeable concept and I think they's kind of smashed together at the moment in a way that suggests reusability but doesn't actually provide it.

Since this PR is already huge, I want to take this as is, where it will always just generate a null table, and work on the other thing seperately. #192 to track.